authors/apps/authentication/tests/test_email_verification.py

```diff
@@ -0,0 +1,64 @@
+import os
+from django.urls import reverse
+from django.core import mail
+from django.conf import settings
+from rest_framework.test import APIClient
+from rest_framework.test import APITestCase
+from rest_framework.views import status
+from ..utils import send_link
+
+
+class TestEmailVerification(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.signup_url = reverse('authentication:auth-register')
+        self.login_url = reverse('authentication:auth-login')
+        self.verify_url = os.getenv('EXPIRED_LINK')
+        self.resend_verify_url = reverse('authentication:auth-reverify')
+
+        self.signup_data = {
+            "user": {
+                "username": "kahara12345",
+                "email": "daveyhash@gmail.com",
+                "password": "H^&lh123d"
+            }}
+        self.login_data = {
+            "user": {
+                "email": "daveyhash@gmail.com",
+                "password": "H^&lh123d"
+            }}
+
+    def test_send_email(self):
+        # Send message.
+        self.assertEqual(len(mail.outbox),0)
+        mail.send_mail(
+            'Authors Haven Verification Link', 'Here is the message.',
+            settings.EMAIL_HOST_USER, ['daveyhash@gmail.com'],
+            fail_silently=False,
+        )
+        # Test that one message has been sent.
+        self.assertEqual(len(mail.outbox), 1)
+        # Verify that the subject of the first message is correct.
+        self.assertEqual(mail.outbox[0].subject, 'Authors Haven Verification Link')
+    
+    def test_expired_verification_link(self):
+        register = self.client.post(self.signup_url,
+                                    self.signup_data,
+                                    format='json')
+        self.assertEqual(register.status_code, status.HTTP_201_CREATED)
+        token = register.data['token']
+        response = self.client.get(self.verify_url, HTTP_AUTHORIZATION='token {}'.format(token))
```

Comment:
I got confused why this was a bad request until i checked the value of`self.verify_url`. The name should reflect the nature of the url.